tools/table_validator/validation.py

```diff
@@ -0,0 +1,447 @@
+            validation_results.append(columns_pivot_to_rows)
+    client.insert_rows(validation_results_table_ref, validation_results)
+    console_link = bigquery_utils.get_console_link_for_table_ref(validation_results_table_ref)
+    print('View Results Table in Console:\n{}'.format(console_link))
+
+
+def create_diff_tables(query_runner, client, config_data, l_table, r_table):
+    l_dataset_name = l_table.split('.')[1]
+    l_table_name = l_table.split('.')[1] + '.' + l_table.split('.')[2]
+
+    r_dataset_name = r_table.split('.')[1]
+    r_table_name = r_table.split('.')[1] + '.' + r_table.split('.')[2]
+
+    destination_dataset = config_data['destinationDataset']
+    destination_table = '{}.{}'.format(destination_dataset, l_table_name.split('.')[1])
+    primary_keys = config_data['primaryKeys']
+
+    columns_list = bigquery_utils.get_full_columns_list(client, config_data['excludeColumnMapping'],
+                                                        config_data['primaryKeys'], l_table_name, r_table_name)
+    if columns_list is None:
+        return None
+    l_column_name = l_table_name.replace('.', '_')
+    r_column_name = r_table_name.replace('.', '_')
+    type_cast_dict = get_type_cast_dict(client, l_table_name, columns_list)
+
+    # Create table containing all mismatches between the columns of the
+    # left and right tables after joining on the primary keys specified in config file
+
+    mismatch_query = generate_query_string_mismatch_row(l_table, r_table,
+                                                        l_column_name, r_column_name,
+                                                        primary_keys, columns_list,
+                                                        type_cast_dict)
+    mismatch_table_name = '{}_mismatches'.format(destination_table)
+
+    # Create table containing the all the missing rows which were present in the left table
+    # but not present in the right table
+    right_table_missing_rows_query = generate_query_string_missing_row(l_table, r_table,
+                                                                       l_column_name, r_column_name,
+                                                                       primary_keys, columns_list, type_cast_dict)
+    right_missing_rows_table_name = '{}_{}_missing_rows'.format(destination_table, r_dataset_name)
+
+    # Create table containing the all the missing rows which were present in the right table
+    # but not present in the left table
+    left_table_missing_rows_query = generate_query_string_missing_row(r_table, l_table,
+                                                                      r_column_name, l_column_name,
+                                                                      primary_keys, columns_list, type_cast_dict)
+    left_missing_rows_table_name = '{}_{}_missing_rows'.format(destination_table, l_dataset_name)
+
+    with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
```

Comment:
Because for the one with 3 workers its only going to have 3 queries running through it at a time so I didn't want to make it have extra workers whereas the other one will have more than 3 and I thought 5 would be a good number of workers to make it not crash I can make them both 5 workers though